### PR TITLE
feat: granular chunking & dynamic routes

### PR DESCRIPTION
**Pull request by jincan39**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [x] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [x] Updated relevant documentation in the code
- [x] Test code changes manually, and describe your procedure in a comment on this PR
- [x] Mark this PR with the correct milestone

## If PR to `main`
- [ ] Update the version number in `package.json` and `src/config/common.json`
- [ ] Update the minimum required signer version number in `src/config/common.json` if applicable
- [ ] Test following the procedure in `docs/manual_test.md`
- [ ] Update `CHANGELOG.md`
- [ ] Make sure that all PR's to `manta-signer` and `sdk` on which this PR depends are merged

### CODE REVIEW

**Comment by jincan39** [comment]
Notes:
 -  For a more elegant approach we can reimplement [Link](https://github.com/vercel/next.js/blob/canary/packages/next/src/client/link.tsx#L533) with IntersectionObserver.
 - Currently Project stick with [webpack v4.44.2](https://github.com/dilanx/craco/blob/v6.4.5/packages/craco/package-lock.json#L21013), for long term we might would like to migrate it to [next.js](https://github.com/vercel/next.js/) which is more pioneering with the support of rust tools chain(e.g.[turbo](https://github.com/vercel/turbo)).

**Review by zhangjing28 — APPROVED** [review]
That's great help, thanks

**Comment by jincan39** [comment]
Notes: 
 - For element-react dynamic, we might could try 
   ``` {
      "plugins": [
          ["transform-modules", {
              "element-ui": {
                   "transform":  "element-react/dist/npm/es5/src/${member}",
                   "kebabCase":  true
              }
          }]
      ]


 - Bundles use eight times of different version of bn.js, use resolution might be risk. https://github.com/indutny/bn.js/issues/273

File: craco.config.js [hunk 1]
```diff
@@ -1,31 +1,115 @@
...
+        'ethers',
+        'element-theme-default',
+        'semantic-ui-css',
+        'semantic-ui-react',
+        'rxjs'
+      ];
+
+      const MANTA_BUNDLES = [
+        'manta.js',
+        'manta.js-kg-dev',
```
Comment (Kevingislason) [inline craco.config.js:47]: Why is this development library included? It is not in the package.json,

Comment (jincan39) [reply craco.config.js:47]: I took this from [branch sbt](https://github.com/Manta-Network/manta-front-end/blob/feature/sbt-demo/package.json#L51) to make it compatible.

**Pull request merged**